pkg/v1/providers/tests/unit/ip_family_test.go

```diff
@@ -525,6 +589,135 @@ var _ = Describe("TKG_IP_FAMILY Ytt Templating", func() {
 		})
 
 	})
+	Describe("infrastructure-vsphere windows overlay.yaml", func() {
+		var paths []string
+		BeforeEach(func() {
+			paths = []string{
+				filepath.Join("fixtures", "yttmocks"),
+				filepath.Join("..", "..", "infrastructure-vsphere", "v1.0.1", "ytt", "overlay-windows.yaml"),
+				filepath.Join("..", "..", "infrastructure-vsphere", "v1.0.1", "ytt", "base-template.yaml"),
+				filepath.Join("..", "..", "config_default.yaml"),
+			}
+		})
+
+		Describe("cluster api server port is configured", func() {
+			var values string
+			When("ip family is configured to ipv4", func() {
+				BeforeEach(func() {
+					values = createDataValues(map[string]string{
+						"CLUSTER_NAME":                "foo",
+						"TKG_CLUSTER_ROLE":            "workload",
+						"TKG_IP_FAMILY":               "ipv4",
+						"CLUSTER_CIDR":                "10.0.0.0/16",
+						"SERVICE_CIDR":                "10.0.0.0/18",
+						"CLUSTER_API_SERVER_PORT":     "443",
+						"IS_WINDOWS_WORKLOAD_CLUSTER": "true",
+					})
+				})
+
+				It("renders control plane template to bind the local apiServer endpoint to '0.0.0.0' as the node IP and port to custom api server port configured", func() {
+					output, err := ytt.RenderYTTTemplate(ytt.CommandOptions{}, paths, strings.NewReader(values))
+					Expect(err).NotTo(HaveOccurred())
+
+					kubeadmControlPlaneDocs, err := FindDocsMatchingYAMLPath(output, map[string]string{
+						"$.kind": "KubeadmControlPlane",
+					})
+					Expect(err).NotTo(HaveOccurred())
+
+					Expect(kubeadmControlPlaneDocs).To(HaveLen(1))
+					Expect(kubeadmControlPlaneDocs[0]).To(HaveYAMLPathWithValue("$.spec.kubeadmConfigSpec.initConfiguration.localAPIEndpoint.advertiseAddress", "0.0.0.0"))
+					Expect(kubeadmControlPlaneDocs[0]).To(HaveYAMLPathWithValue("$.spec.kubeadmConfigSpec.initConfiguration.localAPIEndpoint.bindPort", "443"))
+					Expect(kubeadmControlPlaneDocs[0]).To(HaveYAMLPathWithValue("$.spec.kubeadmConfigSpec.joinConfiguration.controlPlane.localAPIEndpoint.advertiseAddress", "0.0.0.0"))
+					Expect(kubeadmControlPlaneDocs[0]).To(HaveYAMLPathWithValue("$.spec.kubeadmConfigSpec.joinConfiguration.controlPlane.localAPIEndpoint.bindPort", "443"))
+				})
+			})
+			When("ip family is configured to ipv4,ipv6 dual stack settings", func() {
```

Comment:
It's my understanding that `IS_WINDOWS_WORKLOAD_CLUSTER` is only compatible with `TKG_IP_FAMILY` set to `ipv4`. See [these tests](https://github.com/vmware-tanzu/tanzu-framework/blob/5f337991938c1f34b258eef083573541e1624cee/pkg/v1/providers/tests/unit/ip_family_test.go#L123).

(I'd expect these tests to fail as written... do they?)

Perhaps we can just remove these non-ipv4 cases.

Reply:
Since "ytt/03_customizations/ip_family.yaml" is not included in the paths, the compatibility check for windows cluster `ipv6` is not kicking in. I did it on purpose to unit test the changes(in case we support ipv6 for windows in future).

If you think we can remove these non-ipv4 test cases for windows clusters, I am fine with that.

Reply:
updated the PR (removed non-ipv4 cases for windows clusters ).

Reply:
>  I did it on purpose to unit test the changes(in case we support ipv6 for windows in future).

I see. I didn't catch that. Now that I understand I could be convinced either way, but just removing them works for me.  

Thanks for clarifying!